### PR TITLE
Enable Zendesk pre-sales chat in Checkout

### PR DESCRIPTION
**Pull request by yoavf**
#### Proposed Changes

* This enables a Zendesk pre-sales chat widget on checkout (availability of the chat is controlled by the wpcom endpoint)
* It replaces the HappyChat-based pre-sales chat. However, HappyChat will still be available to users who typically have access to it when the ZD isn't available, via the `CheckoutSummaryHelpButton` (as before).

Known issue: Once the ZD loaded widget is loaded, it will stay visible, even if we navigate away from the checkout section. That's currently acceptable.

#### Screenshots
**Zendesk Chat widget active:**
Screenshot 2022-11-17 at 13 44 15

**Zendesk pre-sales chat not available, and user does **not** have chat support level:**
Screenshot 2022-11-17 at 13 43 49

**Zendesk pre-sales chat not available, and user does have access to chat support level:**
Screenshot 2022-11-17 at 13 43 21

#### Testing Instructions

Reproduce the states from the above screenshots in checkout:
 - ZD chat available (by overwriting `presalesZendeskChatAvailable` if it returns false)
 - ZD chat not available, while user does have access to chat support (i,e with an existing plan)
 - ZD chat not available, while user does not have access to chat support

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

This PR partially replaces #69706 - ZD chat in signup will be added separately.

### CODE REVIEW

**Review by sirbrillig — COMMENTED** [review]
I haven't run through the test instructions but this looks good from a Payments perspective. 👍

File: client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx
Comment (DavidRothstein) [inline client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]: It looks like with this code, you'd be taking the pre-sales chat option away from non-English speakers.  Currently with Spanish, for example, I do see a chat option:

pre-sales-chat

I'm not sure if the `isEnglishLocale` check is permanent or temporary, but either way, is there a reason not to leave the old option around as a backup for when Zendesk chat isn't available?

File: client/components/zendesk-chat-widget/index.tsx
Comment (klimeryk) [inline client/components/zendesk-chat-widget/index.tsx]: Nitpick: maybe let's call this `chatKey` to be consistent with `zendeskPresalesChatKey` and `zendesk_presales_chat_key` from the config files. It can also be confused with `sessionId` or some other identifier of the chat (as named in Happy Chat), not the chat widget.

File: client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx
Comment (klimeryk) [reply client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]: > It looks like with this code, you'd be taking the pre-sales chat option away from non-English speakers.

AFAIK, that's the scope of the test for the Zendesk chat widget.
However, if the user is on a non-English locale, they do (from my testing and understanding of the code) get the normal pre-sales chat via Happy Chat option:
Screenshot 2022-11-25 at 13 23 59

Clicking on it goes to Help Center and from there you can contact support (email or chat). Confirmed it's the same for the old contact form (FAB/inline-form):

Screenshot 2022-11-25 at 13 36 05

And the same on production.

Let me know if I misunderstood and you meant a different use case 🙇

Comment (klimeryk) [inline client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]:
```suggestion
	const isPresalesZendeskChatEligible = presalesZendeskChatAvailable && isEnglishLocale && zendeskPresalesChatKey;
```

Should we also check for a truth-y `zendeskPresalesChatKey`? Otherwise, we try to render the `ZendeskChat` component, but only the `div` is rendered, but the `script` tag is not added, so nothing happens and the user might not get the help they need. This won't matter when we enable the key in all envs, but if we deploy it as-is, production users will be affected (during Zendesk Chat hours).

**Review by klimeryk — APPROVED** [review]
Tested with Help Center as well as FAB/inline-form. Looks good to me - except for the possible issue with `zendeskPresalesChatKey`.

The flow of clicking on the action saying "Ask a Happiness Engineer" and then just showing the FAB/Help Center and requiring the user to still click another button to trigger an actual support interaction seems a bit cumbersome, but it's what we seem to have already on production, so maybe it's fine 🤷

File: client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx
Comment (DavidRothstein) [reply client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]: >Clicking on it goes to Help Center and from there you can contact support (email or chat).

That's only if you're **already** a paid customer.  If you're a new customer who hasn't bought anything yet, the Help Center only gives you access to the forums.  Whereas the link shown in my screenshot above is a pre-sales chat option that (when it appears) is currently made available even to users who haven't purchased anything yet.  So that's the thing that this pull request seems like it would be taking away from non-English users.

Comment (kriskarkoski) [reply client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]: I think that difference in behavior might be related to Help Center. If a user is in the test group for that I believe they get the normal channel options in Help Center for their plan, where previously with inline help we were bypassing that and exposing chat even if they didn't have an eligible plan. So I don't think that change in behavior is from this change, but rather the rollout of Help Center. I don't know if it was intended or not cc @Automattic/vertex

Comment (DavidRothstein) [reply client/my-sites/checkout/composite-checkout/components/checkout-help-link.tsx]: This pull request is definitely taking pre-sales chat away from **some** users who currently have it -- that's how I was able to make my screenshot above (that's a user who would lose pre-sales chat after this pull request is merged).

It is definitely possible there is a test going on and some users who are in the test have already lost access to pre-sales chat separately, though... I didn't think about that :smile:  It does seem like I see different behavior on this page with two of my test accounts, and perhaps that's the reason why.

**Comment by elliottprogrammer** [comment]
@yoavf, do you know approximately when you will be deploying this PR? Jetpack is also adding pre-sales Zendesk chat to various pre-sales pages in Jetpack Cloud (Calypso green) and Jetpack.com ( p8wKgj-3em-p2#comment-16965 ), and we need to re-use your `<ZendeskChat />` component that is part of this PR.  We're actually scrambling to get it deployed by EOB today. Here's the PR that utilizes the `<ZendeskChat />` component: https://github.com/Automattic/wp-calypso/pull/70481
If there's something blocking this from shipping, the only other solution I think would be for me to create a separate PR just for the ZendeskChat component, and deploy it. Then for this PR, you would just remove the component code and import the ZendeskChat component instead. What are your thoughts?

**Comment by yoavf** [comment]
> If there's something blocking this from shipping, the only other solution I think would be for me to create a separate PR just for the ZendeskChat component, and deploy it. Then for this PR, you would just remove the component code and import the ZendeskChat component instead. What are your thoughts?

Hi @elliottprogrammer - I saw you closed the other PR, but I don't mind if you'd like to extract the component out of this PR into a new one. We're not ready yet to launch on checkout/signup, so it would make sense to extract it if you'd like to use it sooner.

Edit: oh, I see you've already done that with #70594. I'll rebase then.

**Pull request merged**
